Neos.ContentRepository/Classes/Domain/Utility/NodeContexts.php

```diff
@@ -20,12 +20,12 @@
     /**
      * Determine if the given node path is a context path.
      *
-     * @param string $string path or identifier
+     * @param string $string context path or context identifier
      * @return boolean
      */
     public static function hasContext($string)
     {
-        return (strpos($string, '@') !== false);
+        return (preg_match('/@/u', $string) === 1);
```

Comment:
Why the change to `preg_match` with UTF-8 matching for a single character? IMHO `strpos($string, '@') !== false` is the same as `preg_match('/@/u', $string) === 1`. I'd say we should match for a node path or identifier pattern with added context here. Since this is a public static method we cannot be sure where it will be used so the current implementation is a little bit to naive for me since "Determine if the given node path is a context path." is not really checked here.

(also we need to adjust this doc comment, too)